pandas/util/testing.py

```diff
@@ -868,9 +868,9 @@ def assertIsInstance(obj, cls, msg=''):
     """Test that obj is an instance of cls
     (which can be a class or a tuple of classes,
     as supported by isinstance())."""
-    assert isinstance(obj, cls), (
-        "%sExpected object to be of type %r, found %r instead" % (
-            msg, cls, type(obj)))
+    if not isinstance(obj, cls):
+        err_msg = "{0}Expected type {1}, found {2} instead"
+        raise AssertionError(err_msg.format(msg, cls, type(obj)))
```

Comment:
I think it *would* work right away. But, the power if ``pytest`` is really using fixtures, parameterized tests, and clean looking asserts.

 So need to convert most ``setUp/tearDown`` stuff to fixtures (a lot). properly use fixture scope (e.g. db stuff which you only want to create a database once per test run; this is the most complicated).

Parameterized tests are awesome, but again, need to restructure tests a bit for that.

I suppose the actual assert mechanism could be changed, but a lot of care would be needed.

And here's the rub, I *think* we would have to simply switch, then start changing things as its not back-compat to nose.

Given all this I am not optomistic this could be done quickly/easily. The upside is nice & clean tests, but getting there is hard.